### ISSUE TITLE
how can i use my own weight_pth?

### ISSUE THREAD

**Issue by 1579109909** [kazuto1011/grad-cam-pytorch#30]
thanks to your work for grad-cam!
however how can i use my own weight_pth to product the grad-cam images?

**Comment by kazuto1011** [comment]
Please create your own code like [demo3](https://github.com/kazuto1011/grad-cam-pytorch/blob/master/main.py#L258-L312).
1. Determine [the layer](https://github.com/kazuto1011/grad-cam-pytorch/blob/e8c6abe4938ebfd1fec9a14705a1361cd60707a8/main.py#L278) where you want to visualize activation maps.<br> You can see the available layer names from your weights, e.g., `print(*your_weights.keys(), sep="\n")`.
2. Determine [the class id](https://github.com/kazuto1011/grad-cam-pytorch/blob/e8c6abe4938ebfd1fec9a14705a1361cd60707a8/main.py#L279) to compute gradients.
3. Create a [`GradCAM` instance](https://github.com/kazuto1011/grad-cam-pytorch/blob/e8c6abe4938ebfd1fec9a14705a1361cd60707a8/main.py#L285) with your model.
4. Then `forward()`, `backward()`, `generate()`.

**Comment by ns-kminaga** [comment]
Hi, can you expound on number 3 & 4? @kazuto1011

**Comment by JanineCHEN** [comment]
Thanks a lot for the repo, having same question here, could you kindly elaborate step 3 & 4? @kazuto1011 
Thank you in advance.

**Comment by kazuto1011** [comment]
Suppose you have the model which returns logit in (num_batch, num_class) shape.
Then please create the Grad-CAM instance with the model.
```py
gcam = GradCAM(model=model)
```
The `GradCAM` class has three principal method `forward`, `backward`, and `generate`.

First, please call `forward` with batched images, then you get class probabilities and the corresponding indices in descending order. At the same time, `gcam` saves the intermediate feature maps.
```py
sorted_probs, sorted_ids = gcam.forward(images)
```
Here we take indices of the top 1 classes for example.
```py
top1_ids = sorted_ids[:, [0]]
```
Please call `backward` with the indices. This line computes and saves the class-specific gradients at all layers.
```py
gcam.backward(ids=top1_ids)
```
Now, `gcam` has {feature map, gradient} pairs of all layers.
Finally, please call `generate` with the layer name you want to visualize.
```py
regions = gcam.generate(target_layer='any_layer_names')
```
